### ISSUE TITLE
Multi-Parapgraph Markdown-quote is rendered incorrectly

### ISSUE THREAD

**Issue by pludi** [vector-im/element-web#8786]
### Description

Markdown-quoting of multiple paragraphs into a single quote does not insert paragraphs, but instead renders into a single line.

### Steps to reproduce

In any channel post a multi-paragraph quote, eg.
```
> This is a quote-test with
>
> multiple paragraphs
```

The expected result would be (through `pandoc -f markdown -t html5`)
```
<blockquote>
<p>This is a quote-test with</p>
<p>Multiple paragraphs</p>
</blockquote>
```
Instead the resulting `formatted_body` content is
```
<blockquote>\nThis is a quote-test withMultiple paragraphs\n</blockquote>\n
```

### Version information

- **Server**: Self-hosted, Synapse 0.99.1.1+stretch1
- **Platform**: Web and Desktop

For the web app:

- **Browser**: Chrome 72.0.3626.109
- **OS**: Ubuntu 18.04
- **URL**: riot.im/app

For the desktop app:

- **OS**: Ubuntu 18.0.4
- **Version**: 1.0.1

**Comment by hpdeifel** [comment]
This still happens in riot 1.5.0, so I believe the `type:old-composer` label is incorrect.

**Comment by t3chguy** [comment]
Tested against commonmark dingus and indeed we are doing something wrong

**Comment by PeterNerlich** [comment]
I had a few furious encounters with this recently (I was posting messages and whole emails into a room for the group to see as a quote). One thing I now noticed, it seems to work when the message consists of multiple elements (like a blockquote and normal text) - when it is only a single blockquote, it seems to break

**Comment by jooize** [comment]
This issue drives me crazy as I frequently send messages only consisting of quotes.

**Comment by 4of92000** [comment]
I've run into the same issue and done a bit of experimenting to feel around its edges. What I've found:

## For the formatting to fail, the blockquote must be the only thing in the message.

Text both before it and after it gets rid of this problem.

## The problem does *not* occur when two lines are immediately consecutive.

This:

```markdown
> multi-line
> blockquote
```

puts two lines, as expected:

> multi-line
> blockquote

## The problem *does* occur when there is an empty line between them.

This:

```markdown
> multi-line
> 
> blockquote
```
renders without a break at all:

> multi-lineblockquote

## Mixing the two rules in one message does not affect their behavior.

```markdown
> multi-line blockquote
> both without spaces
> 
> and with them
> and then not again
```
renders as
> multi-line blockquote
> both without spacesand with them
> and then not again

---

Hopefully this information is useful to someone or another.

**Comment by williamkray** [comment]
wow, i'm sad to see how long this has been open. i run into this almost every day.

**Comment by brendanzab** [comment]
To workaround this issue, you can precede a multi-paragraph quote (that is to say, a quote containing quoted blank lines) with something else, either "foo" or a separate empty quote:

```markdown
>

> like
>
> this
```

```md
foo
> or
>
> this
```

Nested quotes can be done by applying this rule again:

```
>

> >
>
> > like
> >
> > this
```

Just noting this down here for next time I encounter this and end up in an infuriating edit-fix loop with #12535. It would be great to see this fixed, it is extremely frustrating to encounter this!
